plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1264,6 +1332,127 @@ private List<ColumnMetadata> getColumnMetadatas(Table table)
                 .collect(toImmutableList());
     }
 
+    @Override
+    public ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSession session, SchemaTableName tableName, Map<String, Object> analyzeProperties)
+    {
+        IcebergTableHandle handle = getTableHandle(session, tableName);
+        if (handle == null) {
+            return null;
+        }
+        checkArgument(handle.getTableType() == DATA, "Cannot analyze non-DATA table: %s", handle.getTableType());
+        verify(handle.getRetryMode() == NO_RETRIES, "Retry mode unexpectedly set: %s", handle.getRetryMode());
+
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, handle.getSchemaTableName());
+        Set<String> allColumnNames = tableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .map(ColumnMetadata::getName)
+                .collect(toImmutableSet());
+
+        Set<String> analyzeColumnNames = getColumnNames(analyzeProperties)
+                .map(columnNames -> {
+                    // validate that proper column names are passed via `columns` analyze property
+                    if (columnNames.isEmpty()) {
+                        throw new TrinoException(INVALID_ANALYZE_PROPERTY, "Cannot specify empty list of columns for analysis");
+                    }
+                    if (!allColumnNames.containsAll(columnNames)) {
+                        throw new TrinoException(
+                                INVALID_ANALYZE_PROPERTY,
+                                format("Invalid columns specified for analysis: %s", Sets.difference(columnNames, allColumnNames)));
```

Comment:
As the column list is not specified at call to `ANALYZE` but is the part of table definition I think this error messages may be a bit confusing. Can you point user to table property?

Reply:
NVM - misread :/